### PR TITLE
refactor(server): Alter routes to be only plugin without wrapper

### PR DESCRIPTION
**Pull request by ff6347**
- Removes wrapper function.
- Removes call to cors on each individual plugin
- fixes async and callback style mixture

What is the benefit of this refactor? We can see now in the build-server function where the plugins are registered.
It removed the error that we mix async and callback code https://app.asana.com/0/1205231177119914/1206862611196241/f

### CODE REVIEW

File: src/lib/build-server.ts [hunk 1]
```diff
@@ -42,44 +48,36 @@ export async function buildServer({
...
+	});
+	server.register(generateAnswerRoute, {
+		prefix: "/generate-answer",
+		OPENAI_MODEL,
+		OPENAI_KEY,
+	});
+	server.register(loadUserRequestRoute, { prefix: "/requests" });
 
-	server.setErrorHandler(function (error, request, reply) {
+	-server.setErrorHandler(function (error, request, reply) {
```
Comment (Jaszkowic) [inline src/lib/build-server.ts:80]: Just for my understanding: What does `-server` mean?

File: src/lib/build-server.ts [hunk 2]
```diff
@@ -42,44 +48,36 @@ export async function buildServer({
...
 	});
-
-	registerDocumentationRoute(server);
-	registerRootRoute(server);
-	registerHealthRoute(server);
-	registerCountDocumentsRoute(server);
+	server.register(fastifySwaggerUi, documentationRouteOptions);
+	server.register(rootRoute);
+	server.register(healthRoute, { prefix: "/health" });
+	// FIXME: We register at /processed_documents and add one handler for /count. Why?
```
Comment (Jaszkowic) [inline src/lib/build-server.ts:62]: I want the route to be `GET /processed_documents/count` because it is more REST-like compared to `GET /count`

**Review by Jaszkowic — APPROVED** [review]
LGTM, tested locally, works fine

File: src/lib/build-server.ts (hunk 1 above)
Comment (ff6347) [reply src/lib/build-server.ts:80]: ups. This is an error. I wounder why this still works and does not fail. JS magic

File: src/lib/build-server.ts (hunk 2 above)
Comment (ff6347) [reply src/lib/build-server.ts:62]: Yes I thought so. This API is not really REST isn't it. It is more like a lot RPCs. Nice thing about having the prefixes there and no other route inside, is that we can see in build-server.ts where all the routes are registered

File: src/lib/build-server.ts (hunk 1 above)
Comment (Jaszkowic) [reply src/lib/build-server.ts]: oh wow. I thought it was some magic operator for error handling :D

**Review by Jaszkowic — APPROVED** [review]

**Pull request merged**